cf-agent/verify_users_pam.c

```diff
@@ -1285,6 +1307,137 @@ static bool DoCreateUser(const char *puser, const User *u, enum cfopaction actio
 
     return true;
 }
+#endif
+
+#ifdef HAVE_PW
+static bool DoCreateUserUsingPw(const char *puser, const User *u, enum cfopaction action,
+                         EvalContext *ctx, const Attributes *a, const Promise *pp)
```

Comment:
There is a lot of duplication here. Can you factor out parts / most of this function to be shared by `DoCreateUserUsingUseradd` ?